virttest/utils_test/libvirt.py

```diff
@@ -3759,3 +3760,278 @@ def check_dumpxml(vm, content, err_ignore=False):
+                                    "swap_hard_limit": "memory.memsw.limit_in_bytes"}
+    __cgroup_v2_mem_file_mapping = {"hard_limit": "memory.max",
+                                    "soft_limit": "memory.high",
+                                    "swap_hard_limit": "memory.swap.max"}
+    __cgroup_v1_schedinfo_file_mapping = {"cpu_shares": "cpu.shares",
+                                          "vcpu_period": "<vcpuX>/cpu.cfs_period_us",
+                                          "vcpu_quota": "<vcpuX>/cpu.cfs_quota_us",
+                                          "emulator_period": "emulator/cpu.cfs_period_us",
+                                          "emulator_quota": "emulator/cpu.cfs_quota_us",
+                                          "global_period": "cpu.cfs_period_us",
+                                          "global_quota": "cpu.cfs_quota_us",
+                                          "iothread_period": "<iothreadX>/cpu.cfs_period_us",
+                                          "iothread_quota": "<iothreadX>/cpu.cfs_quota_us"}
+    __cgroup_v2_schedinfo_file_mapping = {"cpu_shares": "cpu.weight",
+                                          "vcpu_period": "<vcpuX>/cpu.max",
+                                          "vcpu_quota": "<vcpuX>/cpu.max",
+                                          "emulator_period": "emulator/cpu.max",
+                                          "emulator_quota": "emulator/cpu.max",
+                                          "global_period": "cpu.max",
+                                          "global_quota": "cpu.max",
+                                          "iothread_period": "<iothreadX>/cpu.max",
+                                          "iothread_quota": "<iothreadX>/cpu.max"}
+
+    def __init__(self, vm_pid):
+        self.__vm_pid = vm_pid
+
+    def is_cgroup_v2_enabled(self):
+        cmd = "cat /proc/mounts | grep cgroup2"
+        cmd_result = process.run(cmd, ignore_status=True, shell=True)
+        if cmd_result.exit_status:
+            return False
+        return True
+
+    def get_cgroup_path(self, controller=""):
+        cgroup_path = ""
+        if self.is_cgroup_v2_enabled():
+            vm_proc_cgroup_path = "/proc/%s/cgroup" % self.__vm_pid
+            with open('/proc/mounts', 'r') as mnts:
+                cg_mount_point = re.findall(r"\s(\S*cgroup)\s", mnts.read())
+            with open(vm_proc_cgroup_path, 'r') as vm_cg_file:
+                cg_vm_scope = re.findall(r"\S*::(\S*)", vm_cg_file.read())
+            cgroup_path = os.path.join(cg_mount_point[0], cg_vm_scope[0].strip("/"))
+            if "emulator" in cgroup_path:
+                cgroup_path += "/.."
+        else:
+            cgroup_path = utils_cgroup.resolve_task_cgroup_path(int(self.__vm_pid),
+                                                                controller)
+        if not os.path.exists(cgroup_path):
+            logging.error("cgroup path '%s' doesn't exist" % cgroup_path)
```

Comment:
I suppose you need return "" or None here if the path doesn't exit

Reply:
the cgroup_path set as "" in line3815, and it'll be returned in 3830 as "" if not changed.